### PR TITLE
Add `imguiPlugin` sample 

### PR DESCRIPTION
**Pull request by roby2014**
# Description

Added `imguiPlugin` sample

## Checklist

- [X] Self-review changes.
- [X] Evaluate impact on the documentation.

### CODE REVIEW

**Review by RiscadoA — CHANGES_REQUESTED** [review]
Thanks for working on this! Looks good to me :) just some nitpicks + noticed you forgot to add this sample to the list on `docs/pages/3_examples/2_engine/main.md`. Its necessary, as without it the page won't be indexed properly.

image

File: engine/samples/imgui/main.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+using namespace cubos::engine;
+
+/// [ImGui window example]
+static void exampleWindow()
+{
+    ImGui::Begin("Dear ImGui + CUBOS.");
+
+    ImGui::ShowDemoWindow();
+
+    ImGui::End();
```
[the remark this answers is not in the document]
Comment (roby2014) [reply engine/samples/imgui/main.cpp:17]: It's not a problem I think but I can put it outside yeah

**Review by RiscadoA — CHANGES_REQUESTED** [review]
We currently do not include the whole sample code at the end of the examples. Maybe we should, but I think that that should be consistent among all examples.

Some are too long though, so idk how to feel about it. Whats your opinion?

One more thing, if you want the entire file, you do not need the snippet, you can just `@include` it.

**Comment by roby2014** [comment]
> We currently do not include the whole sample code at the end of the examples. Maybe we should, but I think that that should be consistent among all examples.
> 
> Some are too long though, so idk how to feel about it. Whats your opinion?
> 
> One more thing, if you want the entire file, you do not need the snippet, you can just `@include` it.

I always like when docs provide the full example. Also maybe some people just wanna c+p it away. So idk, maybe we can decide this on the next meeting?

File: engine/samples/imgui/page.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# ImGui {#examples-engine-imgui}
+
+@brief Using the @ref imgui-plugin plugin.
+
+The ImGui plugin allows you to integrate the [Dear ImGui](https://github.com/ocornut/imgui) library with CUBOS. engine, making it quick, straightforward, and effortless to create user interfaces and interactive debugging tools within your applications.
```
Comment (DiogoMendonc-a) [inline engine/samples/imgui/page.md:5]: very minor nitpick, but I think it sounds better if we write it either `with Cubos., making it ...` or `with the CUBOS. engine, making it...`

File: engine/samples/imgui/page.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Once the ImGui plugin is added, you can create systems to display ImGui windows and widgets. Here's an example of how to create an ImGui window.
+
+@snippet imgui/main.cpp ImGui window example
+
+Ensure that you add your system with the `cubos.imgui` tag; otherwise, the ImGui elements from that system won't be be visible.
+
+@snippet imgui/main.cpp Adding the system
+
+Pretty simple right? You're now equipped to craft and utilize ImGui's functions to design your cool user interface.
```
Comment (DiogoMendonc-a) [inline engine/samples/imgui/page.md:25]: It'd be nice if we had a link here to where people can check how to use imgui stuff. Someone coming to this sample is probably looking for information that is more in depth than just the process of spawning a window.

**Review by DiogoMendonc-a — CHANGES_REQUESTED** [review]

File: engine/samples/imgui/page.md (hunk 3 above)
Comment (roby2014) [reply engine/samples/imgui/page.md:25]: Well for that we could just add ImGui's documentation link or something.

Comment (DiogoMendonc-a) [reply engine/samples/imgui/page.md:25]: Yeah, that was what I meant, sorry if I wasn't clear, it wasn't telling you to document all of imgui yourself

**Review by DiogoMendonc-a — APPROVED** [review]

**Review by RiscadoA — APPROVED** [review]

**Pull request merged**